packages/yew/src/virtual_dom/mod.rs

```diff
@@ -145,14 +145,22 @@ mod feat_ssr {
     }
 }
 
+/// Defines if the [`Attributes`] is set as element's attribute or property
+#[allow(missing_docs)]
+#[derive(PartialEq, Eq, Copy, Clone, Debug)]
+pub enum ApplyAttributeAs {
+    Attribute,
+    Property,
+}
+
 /// A collection of attributes for an element
 #[derive(PartialEq, Eq, Clone, Debug)]
 pub enum Attributes {
     /// Static list of attributes.
     ///
     /// Allows optimizing comparison to a simple pointer equality check and reducing allocations,
     /// if the attributes do not change on a node.
-    Static(&'static [[&'static str; 2]]),
+    Static(&'static [(&'static str, &'static str, ApplyAttributeAs)]),
```

Comment:
That's for a different PR, imo. We can break the data into separate struct. `&'static str` and `AttrValue::Static` should be the same (after - optimizations) so that can be used. It should also reduce the code complexity